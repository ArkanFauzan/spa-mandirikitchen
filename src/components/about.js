import React from "react"
import {FormAbout} from "./about/form"

export class AboutComponent extends React.Component{
    constructor(){
        super()
        this.state={
            nama:"",
            email:"",
            pesan:"",
        }
    }

    _handleChange = (value) => {
        this.setState({[value.name]:value.value})
    }

    render(){

        const {nama, email, pesan} = this.state

        return(
            <div className="container">
                <img style={{display:"block", margin:"auto"}} width="100%"  src="img/cv-crop.jpg" />
                <h2 className="my-4">CV Mandiri Kitchen Perkasa</h2>
                <p className="mb-4" style={{fontSize:"20px",textAlign:"justify"}}>
                    Mandiri Kitchen Perkasa adalah sebuah perusahaan yang bergerak dibidang 
                    disain dapur restoran serta pembuatan peralatan dapur komersial. Disain 
                    atau perencanaan dapur restoran sangat diperlukan agar ruang dapur 
                    restoran anda menjadi efisien dan fungsional serta biaya yang dikeluarkan 
                    sesuai dengan hasil yg diinginkan karena Penataan Peralatan Memasak yang 
                    TEPAT dan BENAR.  Ruang lingkup dan spesifikasi pekerjaan kami 
                    meliputi Desain dan perencanaan dapur komersial, Pembuatan dan pengadaan 
                    alat alat dapur komersial, instalasi ducting dan instalasi gas restoran. 
                    Yang dimaksud Dapur Komersial disini adalah dapur yang dipakai untuk 
                    keperluan usaha, seperti Dapur Restoran, Dapur Hotel, Dapur Asrama maupun 
                    Dapur Catering</p>

                <h2>Kontak kami</h2>
                <table className="my-4">
                    
                    <tr>
                        <td>Gunadi</td>
                        <td>&nbsp;&nbsp;&nbsp;: 0821 xxxx xxxx</td>
                    </tr>
                    <tr>
                        <td>Heri</td>
                        <td>&nbsp;&nbsp;&nbsp;: 0812 xxxx xxxx</td>
                    </tr>
                    <tr>
                        <td>Website</td>
                        <td>&nbsp;&nbsp;&nbsp;: mandirikitchenperkasa.com</td>
                    </tr>
                    <tr>
                        <td>Email</td>
                        <td>&nbsp;&nbsp;&nbsp;: mandirikitchenperkasa@gmail.com</td>
                    </tr>
                    <tr>
                        <td>Alamat</td>
                        <td>&nbsp;&nbsp;&nbsp;: Depok, Jawa Barat</td>
                    </tr>
                </table>

                <h2>Atau melalui form berikut:</h2>

                {/* menampilkan controlled form dengan prinsip lifting state up */}
                <FormAbout state={this.state} onChangeValue={this._handleChange}/>
            </div>
        )
    }
}