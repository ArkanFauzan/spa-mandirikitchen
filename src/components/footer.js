import React from "react"

export class FooterComponent extends React.Component{
    render(){
        return(
            <>
                <div>
                    <br/>
                    <h3>Hubungi kami</h3>
                    0812 xxxx xxxx (A.N. Gunadi) <br/>0821 xxxx xxxx (A.N. Heri)
                </div><br/>
                <div style={{borderTop:"solid 1px black"}}>
                    <br/>
                    Mandiri Kitchen 2020 all rights reserved
                </div><br/>
            </>
        )
    }
}