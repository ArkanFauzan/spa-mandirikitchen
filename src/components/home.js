import React from "react"
import { Alert, Card, Button
} from 'react-bootstrap';

import {Link, BrowserRouter as Router} from 'react-router-dom'
import {listService} from "./home/service"
import {JumbotronComponenet} from "./home/jumbotron"

export class HomeComponent extends React.Component{
    render(){
        return(
            <Router>
            {/* jumbotron */}
            <JumbotronComponenet />
            
            {/* Card Our service */}
            <div className="row pt-2 bg-light" style={{textAlign:'center'}}>
                {/* menampilkan list service */}
                {listService}

                <div className="my-4 col-12">
                    <Button variant="primary" size="lg" style={{margin:'auto'}}><Link style={{color:'white',textDecoration:'none'}} to="products">Let's See Our Product -&gt;</Link></Button>
                </div>
            </div>

            </Router>
        )
    }
}