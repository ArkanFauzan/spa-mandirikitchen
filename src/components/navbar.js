import React from 'react';
import { Navbar, NavDropdown, Button,
    Nav, Form, FormControl
} from 'react-bootstrap';

import {BrowserRouter as Router,
    Switch,
    Route,
    Link
} from 'react-router-dom'

export function NavbarComponent(){
    return(
        <Navbar fixed="top" bg="light" expand="lg">
            <Navbar.Brand ><Link to="/home"><img style={{marginRight:"50px",}} width="150px" src="img/logo-crop.jpg"/></Link></Navbar.Brand>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
              <Nav className="mr-auto">
                <Nav.Link style={{marginRight:"50px",fontWeight:"bold",textDecoration:"none"}}><Link to="/home">HOME</Link></Nav.Link>
                <Nav.Link style={{marginRight:"50px",fontWeight:"bold",textDecoration:"none"}}><Link to="/products">PRODUCTS</Link></Nav.Link>
                <Nav.Link style={{marginRight:"50px",fontWeight:"bold",textDecoration:"none"}}><Link to="/about">ABOUT</Link></Nav.Link>
              </Nav>
              <Form inline>
                <FormControl type="text" placeholder="Search" className="mr-sm-2" />
                <Button variant="outline-success">Search</Button>
              </Form>
            </Navbar.Collapse>
          </Navbar>
    )
}