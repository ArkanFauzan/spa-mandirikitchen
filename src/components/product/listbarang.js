import React from "react"

export class ListBarang extends React.Component{
    render(){
        const {barang,filter} = this.props;

        // melakukan filter barang
        let barangFilter = barang;
        if(filter!=="semua kategori"){
            barangFilter = barang.filter(value=>{
                return value.kategori===filter
            })
        }

        // membuat list barang berdasarkan barang yang telah difilters
        const list = barangFilter.map((value,index)=>{
            return(
                <div style={{margin:"30px 0px",padding:"0px 10px"}} key={index}>
                    <h3>{index+1}. {value.nama}</h3>
                    <div style={{display:"flex",flexDirection:"column",paddingRight:"20px"}}>
                        <img style={{display:"block",margin:"20px auto", width:"auto"}} src={value.img} />
                        <h4 style={{display:"block",textAlign:"center"}}>{value.deskripsi}</h4>
                        <img style={{display:"block",margin:"20px auto"}} src={value.harga} />
                    </div>
                </div>
            )
        });

        console.log(barangFilter);

        return(
            <>{list}</>
        )
    }
}