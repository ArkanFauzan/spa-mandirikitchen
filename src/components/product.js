import React,{useState, useEffect, useContext} from "react" 
import {BarangContext} from "../barangContext"

// import list barang
import {ListBarang} from "./product/listbarang"

export function ProductComponent(){

    // menggunakan hooks untuk memfilter kategori
    const [state, Filter] = useState({filter:"semua kategori"}) ;
    const {filter} = state;

    // mengambil nilai barang dari provider (menggunakan Context API)
    const {barang} = useContext(BarangContext);

    // mengubah state filter
    const _handleChange = (event) => {
        Filter({filter:event.target.value})
    }

    // similar componentDidUpdate(){}
    useEffect(()=>{
        alert(`Produk yang ditampilkan berdasarkan kategori ${filter}`)
    },[filter])

    return(
        <div className="container-fluid" style={{marginTop:"100px"}}>
            <form>
                <label><p style={{fontSize:"25px",textAlign:"justify"}}>Tampilkan berdasarkan : </p></label>
                <select  style={{fontSize:"25px",padding:"7px 15px"}} name="filter" value={filter} onChange={_handleChange}>
                    <option value="semua kategori">Semua Kategori</option>
                    <option value="area memasak">Area Memasak</option>
                    <option value="area mencuci">Area Mencuci</option>
                </select>
            </form>

            {/* menampilkan list barang sesuai filter kategori */}
            <ListBarang barang={barang} filter={filter}/>
        </div>
    )
}