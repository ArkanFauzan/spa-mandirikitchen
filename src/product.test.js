import React from 'react';
import {ProductComponent} from './components/product';
import {ListBarang} from './components/product/listbarang';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

const barang = {barang:[
    {
        kategori:"area memasak",
        nama:"Exhaust Hood w/ Filter dan tidak dengan Filter, termasuk lampu penerangan",
        img: "img/1.jpg",
        deskripsi:"List daftar harga per-meternya adalah sebagai berikut:",
        harga:"img/1a.jpg"
    },
    {
        kategori:"area memasak",
        nama:"Kwali Range Single, Double dan Triple",
        img: "img/2.jpg",
        deskripsi:"List daftar harga untuk kwali range adalah sebagai berikut: ",
        harga:"img/2a.jpg"
    },
    {
        kategori:"area mencuci",
        nama:"Solid dan Slotted Pan Rack 4 Tier",
        img: "img/10.jpg",
        deskripsi:"List daftar harga per-meternya adalah sebagai berikut: ",
        harga:"img/10a.jpg"
    },
    {
        kategori:"area mencuci",
        nama:"Sink",
        img: "img/11.jpg",
        deskripsi:"List daftar harga Sink adalah sebagai berikut : ",
        harga:"img/11a.jpg"
    },
  ]}

it('Product is renders correctly', () => {
    let component = renderer.create(<ProductComponent/>);  
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

    tree.props._handleChange;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.useEffect;
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const { getByText } = render(<ListBarang barang={barang.barang} filter="semua kategori" />);
    const linkElement = getByText(/Filter dan tidak dengan Filter, termasuk lampu penerangan/i);
    expect(linkElement).toBeInTheDocument();
  });
