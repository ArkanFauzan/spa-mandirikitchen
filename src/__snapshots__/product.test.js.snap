// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product is renders correctly 1`] = `
<div
  className="container-fluid"
  style={
    Object {
      "marginTop": "100px",
    }
  }
>
  <form>
    <label>
      <p
        style={
          Object {
            "fontSize": "25px",
            "textAlign": "justify",
          }
        }
      >
        Tampilkan berdasarkan : 
      </p>
    </label>
    <select
      name="filter"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "25px",
          "padding": "7px 15px",
        }
      }
      value="semua kategori"
    >
      <option
        value="semua kategori"
      >
        Semua Kategori
      </option>
      <option
        value="area memasak"
      >
        Area Memasak
      </option>
      <option
        value="area mencuci"
      >
        Area Mencuci
      </option>
    </select>
  </form>
</div>
`;

exports[`Product is renders correctly 2`] = `
<div
  className="container-fluid"
  style={
    Object {
      "marginTop": "100px",
    }
  }
>
  <form>
    <label>
      <p
        style={
          Object {
            "fontSize": "25px",
            "textAlign": "justify",
          }
        }
      >
        Tampilkan berdasarkan : 
      </p>
    </label>
    <select
      name="filter"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "25px",
          "padding": "7px 15px",
        }
      }
      value="semua kategori"
    >
      <option
        value="semua kategori"
      >
        Semua Kategori
      </option>
      <option
        value="area memasak"
      >
        Area Memasak
      </option>
      <option
        value="area mencuci"
      >
        Area Mencuci
      </option>
    </select>
  </form>
</div>
`;

exports[`Product is renders correctly 3`] = `
<div
  className="container-fluid"
  style={
    Object {
      "marginTop": "100px",
    }
  }
>
  <form>
    <label>
      <p
        style={
          Object {
            "fontSize": "25px",
            "textAlign": "justify",
          }
        }
      >
        Tampilkan berdasarkan : 
      </p>
    </label>
    <select
      name="filter"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "25px",
          "padding": "7px 15px",
        }
      }
      value="semua kategori"
    >
      <option
        value="semua kategori"
      >
        Semua Kategori
      </option>
      <option
        value="area memasak"
      >
        Area Memasak
      </option>
      <option
        value="area mencuci"
      >
        Area Mencuci
      </option>
    </select>
  </form>
</div>
`;
